src/cache/rex_cache.go

```diff
@@ -0,0 +1,59 @@
+// +build !bootstrap
+
+// Cache based on the Google remote execution API.
+// This will likely supersede the RPC cache at some point in the future.
+
+package cache
+
+import (
+	"encoding/hex"
+
+	"github.com/thought-machine/please/src/core"
+	"github.com/thought-machine/please/src/remote"
+)
+
+type rexCache struct {
+	client *remote.Client
+}
+
+func newRemoteCache(state *core.BuildState) *rexCache {
+	return &rexCache{client: remote.Get(state)}
+}
+
+func (rc *rexCache) Store(target *core.BuildTarget, key []byte, files ...string) {
+	log.Debug("Storing %s in remote cache...", target.Label)
+	if err := rc.client.Store(target, key, cacheArtifacts(target, files...)); err != nil {
+		log.Warning("Error storing artifacts in remote cache: %s", err)
+	}
+}
+
+func (c *rexCache) StoreExtra(target *core.BuildTarget, key []byte, file string) {
+	log.Debug("Storing %s: %s in remote cache...", target.Label, file)
```

Comment:
~Missing an implementation here?~ Ah, I see there's no way to retrieve these - maybe remove the log line?

Reply:
Yeah I realised after this line that I'd need to do a refactor of the whole API for this (which I think will be a bit neater overall though). I'll take the message out for now.